src/internal/partial_solution.rs

```diff
@@ -17,6 +17,12 @@ use crate::{
     solver::DependencyProvider,
 };
 
+#[derive(Clone)]
+struct AssignmentDepth<P: Package, V: Version> {
```

Comment:
Maybe `AssignmentWithDecisionLevel` would be more adequate even though a bit redundant with its content. `AssignmentDepth` sounds like something it's not. Is it the fact that `usize` does not convey a lot of meaning that bothers you? Maybe a type alias would be enough here?

Reply:
> Maybe AssignmentWithDecisionLevel would be more adequate even though a bit redundant with its content. 

I can rename that no problem. A bit verbose, but can't come up with anything better. `DecidedAssignment`? :smile: 

> Is it the fact that usize does not convey a lot of meaning that bothers you?

I like using type system to indicate meaning and make it easily greppable. Allows for easy type changes as well: say we want to reduce `u32` to `u16`, it would simplify doing such refactoring a lot.

> Maybe a type alias would be enough here?

Let's see, type alias also provides some of those benefits. There is a problem that you could forget to use it, so searching usages becomes less reliable. Even if you never forget new instance creation becomes hidden (just 0/1 in code). It also can not guard against adding that and something else.
As for benefits... Hm... Both are fully optimized away, so I don't know. Why do you like it more?

Reply:
> `DecidedAssignment`

`AppliedAssignment` as another option.

> Maybe a type alias would be enough here?

Wait, I see another thread talking about type alias/newtype for `DecisionLevel`. Were you asking about the whole `AssignmentDepth` here? If so, I got confused by "Is it the fact that usize does not convey a lot of meaning that bothers you?" question.

Let's see... It keeps return types in functions the same as newtype, quite readable. I'm only concerned about matching on it: you'd have to look up the type definition and do tuple destructuring in that case, while newtype allows field access by name.